src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -37,6 +42,10 @@ bool to_key(streamer& str, const T& tokey, ddsi_keyhash_t& hash)
   std::vector<unsigned char> buffer(sz + padding);
   memset(buffer.data() + sz, 0x0, padding);
   str.set_buffer(buffer.data());
+  /* TODO: what is key endianness to be used here?
+   * since, this may be different between nodes, and if this value is used
+   * for global lookups or the like, this
+   * may cause discrepancies. */
```

Comment:
I'd say, if streams can be both big and little endian, we should settle on what prefer to write out and use that. Seems like something @eboasson might want to weigh in on and then document it somewhere.

Reply:
Always write out native endianness (most of the time the endianness is the same throughout the network) so that's definitely the fastest.

(There are exceptions: a DDSI keyhash is based on big-endian, but fortunately those are hardly ever needed. Furthermore DDS Security requires some discovery data to be little-endian, but that's not relevant to a language binding. The OMG is certainly consistent in being inconsistent ...)